geo-generic-alg/src/algorithm/line_measures/metric_spaces/euclidean/distance.rs

```diff
@@ -384,6 +384,398 @@ fn ring_contains_coord<T: GeoNum>(ring: &LineString<T>, c: Coord<T>) -> bool {
+    fn distance_ext(&self, metric_space: &impl Distance<F, Point<F>, Point<F>>, other: &Self) -> F;
+}
+
+/// Helper function to convert WKB geometry to concrete Point for distance calculation
+fn extract_point_from_wkb<F: CoordFloat, G: PointTraitExt<T = F>>(geom: &G) -> Option<Point<F>> {
+    geom.coord().map(|coord| Point::new(coord.x(), coord.y()))
+}
+
+/// Helper function to convert WKB geometry to concrete LineString for distance calculation
+fn extract_linestring_from_wkb<F: CoordFloat, G: LineStringTraitExt<T = F>>(
+    geom: &G,
+) -> LineString<F> {
+    let coords: Vec<_> = geom
+        .coords_ext()
+        .map(|coord| Coord {
+            x: coord.x(),
+            y: coord.y(),
+        })
+        .collect();
+    LineString::new(coords)
+}
+
+/// Helper function to convert WKB geometry to concrete Polygon for distance calculation
+fn extract_polygon_from_wkb<F: CoordFloat, G: PolygonTraitExt<T = F>>(
+    geom: &G,
+) -> Option<Polygon<F>> {
+    let exterior = geom.exterior()?;
+    let exterior_coords: Vec<_> = exterior
+        .coords()
+        .map(|coord| Coord {
+            x: coord.x(),
+            y: coord.y(),
+        })
+        .collect();
+
+    let holes: Vec<_> = geom
+        .interiors()
+        .map(|interior| {
+            interior
+                .coords()
+                .map(|coord| Coord {
+                    x: coord.x(),
+                    y: coord.y(),
+                })
+                .collect::<Vec<_>>()
+        })
+        .map(|coords| LineString::new(coords))
+        .collect();
+
+    Some(Polygon::new(LineString::new(exterior_coords), holes))
+}
+
+// Implementation of DistanceExt for WKB geometries using the type-tag pattern
+impl<F, G> DistanceExt<F> for G
+where
+    F: GeoFloat, // Use GeoFloat to support polygon operations
+    G: GeoTraitExtWithTypeTag + WkbDistanceTrait<F, G::Tag>,
+{
+    fn distance_ext(&self, metric_space: &impl Distance<F, Point<F>, Point<F>>, other: &Self) -> F {
+        self.wkb_distance_trait(metric_space, other)
+    }
+}
+
+// Internal trait that dispatches to appropriate distance calculation based on geometry types
+trait WkbDistanceTrait<F, GT: GeoTypeTag>
+where
+    F: GeoFloat,
+{
+    fn wkb_distance_trait(
+        &self,
+        metric_space: &impl Distance<F, Point<F>, Point<F>>,
+        other: &Self,
+    ) -> F;
+}
+
+// Point-to-Point WKB distance using existing metric space implementation
+impl<F, P: PointTraitExt<T = F>> WkbDistanceTrait<F, PointTag> for P
+where
+    F: GeoFloat,
+{
+    fn wkb_distance_trait(
+        &self,
+        metric_space: &impl Distance<F, Point<F>, Point<F>>,
+        other: &Self,
+    ) -> F {
+        if let (Some(p1), Some(p2)) = (extract_point_from_wkb(self), extract_point_from_wkb(other))
+        {
+            // Use existing Point-to-Point distance implementation
+            metric_space.distance(p1, p2)
+        } else {
+            F::zero()
+        }
+    }
+}
```

Comment:
Why all these things has WKB in their names? This trait should be general and work with any types implementing geo-trait-ext. There should be nothing specific about WKB.